modules/thunder_updater/thunder_updater.install

```diff
@@ -5,23 +5,105 @@
  * Thunder Updater install hooks.
  */
 
-use Drupal\Core\Extension\MissingDependencyException;
+use Drupal\Core\Entity\EntityStorageException;
+use Drupal\update_helper_checklist\Entity\Update;
 
 /**
- * Install the thunder updater.
+ * Switch to update_helper module instead of thunder_updater.
  */
-function thunder_updater_install() {
-  $logger = \Drupal::logger('thunder_updater');
+function thunder_updater_update_8001() {
+  /** @var \Drupal\Core\Extension\ModuleHandlerInterface $module_handler */
+  $module_handler = \Drupal::moduleHandler();
 
-  // Install Checklist API as soft dependency.
-  try {
-    \Drupal::service('module_installer')->install(['checklistapi']);
-    \Drupal::service('thunder_updater.update_checklist')->markAllUpdates();
-  }
-  catch (MissingDependencyException $e) {
-    $logger->warning(
-      'Could not install Checklist API module. You will not see information 
-      about updates in the drupal admin pages.'
-    );
+  /** @var \Drupal\Core\Extension\ModuleInstallerInterface $module_installer */
+  $module_installer = \Drupal::service('module_installer');
+
+  /** @var \Drupal\Core\Config\ConfigFactoryInterface $config_factory */
+  $config_factory = \Drupal::configFactory();
+
+  // We let exception bubble in this case, since it's not possible to continue
+  // with execution of update hooks without "update_helper.updater" service.
+  $module_installer->install(['update_helper']);
+
+  /** @var \Drupal\update_helper\Updater $updater */
+  $updater = \Drupal::service('update_helper.updater');
+  $update_logger = $updater->logger();
+
+  // We need database connection in order to migrate Thunder updates correctly.
+  $database = \Drupal::database();
+
+  // Existing module states should be preserved!
+  $has_thunder_updater_checklist = $module_handler->moduleExists('checklistapi');
+
+  // Migrate existing Thunder update entities and checklist config.
+  if ($has_thunder_updater_checklist) {
+    try {
+      $module_installer->install(['update_helper_checklist']);
+
+      /** @var \Drupal\update_helper_checklist\UpdateChecklist $update_checklist */
+      $update_checklist = \Drupal::service('update_helper_checklist.update_checklist');
+      $update_checklist->markAllUpdates(FALSE);
+
+      $update_checklist->markUpdatesSuccessful(['thunder' => ['thunder_updater_update_8001']]);
+    }
+    catch (Exception $e) {
+      $update_logger->warning('Unable to install "update_helper_checklist".');
+    }
+
+    // We do not need entities created on "update_helper_checklist" install.
```

Comment:
Why is that? What kind of entities are on that list?

Reply:
When `update_helper_checklist` module is installed, all checklist entries are marked as checked and that action creates an `Update` entity for every checklist entry. That's why we have to remove them and then migrate the current status.

I've changed comment a bit in order to be more understandable.